crates/biome_js_analyze/src/lint/nursery/no_label_without_control.rs

```diff
@@ -0,0 +1,290 @@
+    /// - Wraps an `input` element (or an indicated custom component that will output an `input` element)
+    /// - Has a `for` or `htmlFor` attribute and that the `label` element/component has accessible text content.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label for="js_id" />;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label for="js_id"><input /></label>;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label htmlFor="js_id" />;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label htmlFor="js_id"><input /></label>;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label>A label</label>;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <div><label /><input /></div>;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// <label for="js_id" aria-label="A label" />;
+    /// ```
+    ///
+    /// ```jsx
+    /// <label for="js_id" aria-labelledby="A label" />;
+    /// ```
+    ///
+    /// ```jsx
+    /// <label htmlFor="js_id" aria-label="A label" />;
+    /// ```
+    ///
+    /// ```jsx
+    /// <label htmlFor="js_id" aria-labelledby="A label" />;
+    /// ```
+    ///
+    /// ```jsx
+    /// <label>A label<input /></label>;
+    /// ```
+    ///
+    /// ```jsx
+    /// <label>A label<textarea /></label>;
+    /// ```
+    ///
+    /// ```jsx
+    /// <label><img alt="A label" /><input /></label>;
+    /// ```
+    ///
+    /// ## Options
+    ///
+    /// The rule supports the following options:
+    /// - `controlComponents` - An array of component names that should be considered the same as an `input` element.
+    /// - `labelAttributes` - An array of attributes that should be treated as the `label` accessible text content.
+    /// - `labelComponents` - An array of component names that should be considered the same as a `label` element.
+    ///
+    /// ```json
+    /// {
+    ///     "//": "...",
+    ///     "options": {
+    ///         "controlComponents": ["CustomInput"],
+    ///         "labelAttributes": ["label"],
+    ///         "labelComponents": ["CustomLabel"]
+    ///     }
+    /// }
+    /// ```
+    ///
+    pub NoLabelWithoutControl {
+        version: "next",
+        name: "noLabelWithoutControl",
+        language: "jsx",
+        sources: &[RuleSource::EslintJsxA11y("label-has-associated-control")],
+        recommended: false,
+    }
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct NoLabelWithoutControlOptions {
+    pub label_components: Vec<String>,
+    pub label_attributes: Vec<String>,
+    pub control_components: Vec<String>,
+}
```

Comment:
Options must be documented using doc comments. These comments will be part of the JSON schema, which our users will use.

Reply:
oh, so that's how it works, good to know, I tried to manually add the description on the schema JSON but it got wiped out 😅